src/sentry/integrations/tasks/sync_status_inbound.py

```diff
@@ -0,0 +1,261 @@
+                                {"resolved_in_release": resolved_in_release.version}
+                            )
+                            logger.info(
+                                "get_resolutions_and_activity_data_for_groups.resolved_in_release",
+                                extra=local_logging_params,
+                            )
+                        except Release.DoesNotExist:
+                            # If it gets here, it means we don't know the upcoming
+                            # release yet because it does not exist, and so we should
+                            # fall back to our current model
+                            logger.info(
+                                "get_resolutions_and_activity_data_for_groups.next_release_not_found",
+                                extra=local_logging_params,
+                            )
+
+                resolutions_by_group_id[group.id] = resolution_params
+
+    return resolutions_by_group_id, activity_type, activity_data
+
+
+@instrumented_task(
+    name="sentry.integrations.tasks.sync_status_inbound",
+    queue="integrations",
+    default_retry_delay=60 * 5,
+    max_retries=5,
+    silo_mode=SiloMode.REGION,
+)
+@retry(exclude=(Integration.DoesNotExist,))
+@track_group_async_operation
+def sync_status_inbound(
+    integration_id: int, organization_id: int, issue_key: str, data: Mapping[str, Any]
+) -> None:
+    from sentry.integrations.mixins import ResolveSyncAction
+
+    integration = integration_service.get_integration(integration_id=integration_id)
+    if integration is None:
+        raise Integration.DoesNotExist
+
+    organizations = Organization.objects.filter(id=organization_id)
+    affected_groups = Group.objects.get_groups_by_external_issue(
+        integration, organizations, issue_key
+    )
+    if not affected_groups:
+        return
+
+    installation = integration.get_installation(organization_id=organization_id)
+    assert (
+        installation and installation.org_integration
+    ), "Installation  and org_integration must exist to get a config"
+    config = installation.org_integration.config
+
+    try:
+        assert hasattr(
+            installation, "get_resolve_sync_action"
+        ), "installation be a IssueSyncMixin to get a resolve sync action"
+        # This makes an API call.
+        action = installation.get_resolve_sync_action(data)
+    except Exception:
+        return
+
+    provider = installation.model.get_provider()
+    activity_data = {
+        "provider": provider.name,
+        "provider_key": provider.key,
+        "integration_id": integration_id,
+    }
+    if action == ResolveSyncAction.RESOLVE:
+        (
+            resolutions_by_group_id,
+            activity_type,
+            activity_data,
+        ) = get_resolutions_and_activity_data_for_groups(
+            list(affected_groups),
+            str(config.get("resolution_strategy")),
```

Comment:
coerced for mypy

Reply:
`str(x.get(...))` is going to end up with `"None"` which is almost certainly wrong -- probably the correct response to this error should be `config['resolution_strategy']` -- or the caller should accept `None`?

Reply:
gave caller `None`typehint for this param 👍